File: level80/classwork/codwars4.py
#https://www.codewars.com/kata/635fc0497dadea0030cb7936/train/python
def count_different_matrices(matrices):
    def rotations(mat):
        a, b, c, d = mat
        return [
            [a, b, c, d], 
            [c, a, d, b],    
            [d, c, b, a],    
            [b, d, a, c]      
        ]
    
    seen = set()
    for mat in matrices:
        normalized = min(rotations(mat))
        seen.add(tuple(normalized))
    
    return len(seen)

File: level 080/classwork/codwars4.js
//https://www.codewars.com/kata/635fc0497dadea0030cb7936/train/javascript
const countDifferentMatrices = matrices => {
  const isNew = (seen => ([a, b, c, d]) =>
    (str => ![0, 1, 2, 3].some(i => str.slice(i) + str.slice(0, i) in seen)
               &&
             (seen[str] = true)
    )('' + a + b + d + c))({});
  
  return matrices.filter(isNew).length;
}